### PR TITLE
Implemented copyWith method as per issue #74613 

### LINKED ISSUE #74613: BorderRadius.copyWith is missing

### PR DESCRIPTION
**Pull request by thisisamank**
Earlier copyWith method wasn't present in BorderRadius class so Implemented BorderRadius.copyWith method and it fixed the issue #74613

Fixes #74613

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [x] I listed at least one issue that this PR fixes in the description above.
- [x] I added new tests to check the change I am making or feature I am adding, or Hixie said the PR is test exempt.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] I signed the [CLA].
- [x] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[Features we expect every widget to implement]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#features-we-expect-every-widget-to-implement
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat

### CODE REVIEW

File: packages/flutter/test/painting/border_radius_test.dart [hunk 1]
```diff
@@ -536,4 +536,17 @@ void main() {
     expect((a.add(b.subtract(a) * 0.0)).resolve(TextDirection.ltr), a);
     expect((a.add(b.subtract(a) * 1.0)).resolve(TextDirection.rtl), b.resolve(TextDirection.rtl));
   });
+  test('BorderRadius copyWith', () {
```
Comment (HansMuller) [inline packages/flutter/test/painting/border_radius_test.dart:539]: A test of the basics should be included:
```dart
  test('BorderRadius copyWith, merge, ==, hashCode basics', () {
    expect(const BorderRadius(), const BorderRadius().copyWith());
    expect(const BorderRadius().hashCode, const BorderRadius().copyWith().hashCode);
  });
```

File: packages/flutter/lib/src/painting/border_radius.dart [hunk 2]
```diff
@@ -327,6 +327,21 @@ class BorderRadius extends BorderRadiusGeometry {
     this.bottomRight = Radius.zero,
   });
 
+  /// Creates a border radius by overriding the previous values and
```
Comment (HansMuller) [inline packages/flutter/lib/src/painting/border_radius.dart:330]: There's boilerplate for this comment:
```dart  
/// Returns a copy of this BorderRadius with the given fields replaced with
/// the new values.
```

File: packages/flutter/lib/src/painting/border_radius.dart [hunk 3]
```diff
@@ -327,6 +327,21 @@ class BorderRadius extends BorderRadiusGeometry {
...
+  /// returing a new object
+  BorderRadius copyWith({
+    Radius? topLeft,
+    Radius? topRight,
+    Radius? bottomLeft,
+    Radius? bottomRight,
+  }) {
+    return BorderRadius.only(
+      topLeft: topLeft ?? _topLeft,
```
Comment (HansMuller) [inline packages/flutter/lib/src/painting/border_radius.dart:339]: There's boilerplate for this too:
```dart
topLeft: topLeft ?? this.topLeft,
topRight: topRight ?? this.topRight,
bottomLeft: bottomLeft ?? this.bottomLeft,
bottomRight: bottomRight ?? this.bottomRight,
```

File: packages/flutter/lib/src/widgets/scrollable.dart [hunk 4]
```diff
@@ -980,7 +980,8 @@ class ScrollAction extends Action<ScrollIntent> {
   bool isEnabled(ScrollIntent intent) {
     final FocusNode? focus = primaryFocus;
     final bool contextIsValid = focus != null && focus.context != null;
-    if (contextIsValid) {
+    //had to check for focus!=null because it was giving error
+    if (contextIsValid && focus!=null) {
```
Comment (HansMuller) [inline packages/flutter/lib/src/widgets/scrollable.dart:984]: This change doesn't belong in this PR

File: packages/flutter/test/painting/border_radius_test.dart [hunk 5]
```diff
@@ -536,4 +536,17 @@ void main() {
...
+      topLeft: Radius.elliptical(10.0, 20.0),
+      topRight: Radius.elliptical(30.0, 40.0),
+      bottomLeft: Radius.elliptical(50.0, 60.0),
+      bottomRight: Radius.elliptical(40.0, 80.0)
+    );
+    final BorderRadius secondRadius = firstRadius.copyWith(bottomRight: Radius.zero);
+    expect(firstRadius.topLeft, secondRadius.topLeft);
+    expect(firstRadius.topRight, secondRadius.topRight);
+    expect(firstRadius.bottomLeft, secondRadius.bottomLeft);
+    expect(firstRadius.bottomRight, isNot(secondRadius.topLeft));
```
Comment (HansMuller) [inline packages/flutter/test/painting/border_radius_test.dart:550]: If we're going to test one parameter, we should check them all.
```dart
test('BorderRadius copyWith parameters', () {
  const Radius radius = Radius.circular(10));
  const borderRadius = BorderRadius.all(radius);
  expect(borderRadius.copyWith(topLeft: Radius.zero).topLeft, Radius.zero);
  expect(borderRadius.copyWith(topLeft: Radius.zero).copyWith(topLeft: radius), borderRadius);
  expect(borderRadius.copyWith(topRight: Radius.zero).topRight, Radius.zero);
  ...
});
```

**Review by HansMuller — APPROVED** [review]
LGTM

**Comment by HansMuller** [comment]
@thisisamank - I'm afraid this PR needs to be rebased with the latest master to avoid the test failures (unrelated to your changes). Could you push an update?

After you fetch/rebase you can just push an empty commit to get the tests restarted: `git commit --allow-empty -m 'Empty commit to force all tests to rerun'`

**Comment by thisisamank** [comment]
@HansMuller i don't know why `flutter-build` test is failing all the time

**Comment by HansMuller** [comment]
I'm not sure what the trouble is, it looks like you applied to appropriate remedy. I'm going to try and land this change via a new PR - https://github.com/flutter/flutter/pull/75340. Sorry about all of the hassle.
